cupy/fft/_callback.pyx

```diff
@@ -337,8 +346,9 @@ cdef class _CallbackManager:
             os.makedirs(cache_dir, exist_ok=True)
         path = os.path.join(cache_dir, mod_filename)
         if not os.path.isfile(path):
-            # Set up temp directory
-            tempdir_obj = tempfile.TemporaryDirectory()
+            # Set up temp directory; it must be under the cache directory so
+            # that atomic moves within the same filesystem can be guaranteed
+            tempdir_obj = tempfile.TemporaryDirectory(dir=cache_dir)
```

Comment:
super pedantic here, how about a context manager?

Reply:
I guess I didn't use context manager because the indentation level is already deep (=3). If an error happens, `TemporaryDirectory` guarantees it's cleaned up.